%% LyX 2.0.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english]{article}
\usepackage{mathptmx}
\usepackage{helvet}
\usepackage{courier}
\renewcommand{\familydefault}{\rmdefault}
\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.54cm,bmargin=2.54cm,lmargin=2.54cm,rmargin=2.54cm}
\usepackage{textcomp}
\usepackage{graphicx}

\makeatletter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LyX specific LaTeX commands.
\DeclareRobustCommand{\greektext}{%
  \fontencoding{LGR}\selectfont\def\encodingdefault{LGR}}
\DeclareRobustCommand{\textgreek}[1]{\leavevmode{%
  \IfFileExists{grtm10.tfm}{}{\fontfamily{cmr}}\greektext #1}}
\DeclareFontEncoding{LGR}{}{}
\DeclareTextSymbol{\~}{LGR}{126}

\makeatother

\usepackage{babel}
\begin{document}

\title{First semester final report}


\author{Colin Diesh}


\date{Senior Thesis Project, Fall 2012}
\maketitle
\begin{abstract}
Transcription factors are proteins which recognize short DNA sequences
and they can regulate gene expression by binding to DNA. Using chromatin
immunoprecipitation and high throughput sequencing (ChIP-seq) we can
identify transcription factor binding sites by finding peaks in the
data: places where there is are many DNA reads from immunoprecipitation.
Comparing ChIP-seq data from multiple experiments presents many logistical
challenges -- different ChIP-seq experiments have may have different
signal/noise ratios and using control experiments and normalization
is important. Comparing transcription factor binding sites from different
genomes also presents challenges because binding sites can vary significantly
due to modifications of DNA binding sites, etc.

Previous methods to identify conserved transcription factor binding
sites analyze different genomes separately, then look for binding
sites in conserved and aligned locations {[}1{]}. We use a model for
comparing transcription factor binding from multiple genomes by analyzing
a normalized difference score that uses information from multiple
ChIP-seq experiments. We propose that this method can help find conserved
binding sites across multiple experiments. We evaluated several existing
techniques for analyzing ChIP-seq data analysis and propose our method
for comparing of ChIP-seq data.
\end{abstract}

\section{Background}


\subsection{ChIP-seq basics}

The ChIP-seq protocol uses high throughput sequencing and chromatin
immunoprecipitation (ChIP) to analyze protein-DNA interactions. Antibodies
for immunoprecipitation have been developed to analyze transcription
factors, histones, and DNA methylation (meDIPSeq). Another ChIP method
uses oligonucleotide arrays for tiling the genome (ChIP-chip) instead
of high throughput sequencing. ChIP-seq does not require construction
of tiling arrays and so, instead, the analysis of high throughput
sequence data is the main focus.\\



\subsection{Odom, Dowell et al. - Tissue specific transcriptional regulation
has diverged significantly between human and mouse}
\begin{enumerate}
\item Odom, Dowell et al. used ChIP-chip tiling arrays to analyze binding
site differences between human and mouse. 
\item They found orthologous genes in human and mouse and created tiling
arrays from the gene promoter regions
\item They proposed two approaches for data anlysis

\begin{enumerate}
\item - Gene centric approach for ChIP

\begin{enumerate}
\item Classify binding as true/false for each gene (tiling array as a whole)
\item Similar to Zheng et al. by identifying target genes 
\end{enumerate}
\item - Peak centric approach for analysis

\begin{enumerate}
\item Classify peaks within tiling array
\item Find 'peaks' using ChIP-chip and compare aligned sites
\item Classify peaks as conserved, turnover, gain/loss, or unaligned\\
\includegraphics[scale=0.6]{peak_classification}
\item Odom, Dowell results Fig 2c. (Shared binding site patterns)

\begin{enumerate}
\item (1) Alignment in humans/mouse exists and binding sites are aligned
\textasciitilde{}30\%
\item (2) Alignment in human/mouse exists but binding site are not aligned
\textasciitilde{}30\%
\item (3) No alignment in human/mouse exists but binding site still shared
\textasciitilde{}30\%
\end{enumerate}
\item Conclusion: many shared binding sites are not aligned (60\%+) 
\end{enumerate}
\end{enumerate}
\item Questions about Odom, Dowell

\begin{enumerate}
\item How can we use their results to answer questions about ChIP-seq where
sequence comparison is important for ChIP-seq sequence similarity?
\end{enumerate}
\end{enumerate}

\subsection{Zheng et al. \textendash{} Genetic analysis of transcription factor
binding variation in yeast}
\begin{enumerate}
\item Zheng et al. analyzed binding sites from many yeast ChIP-experiments
using and found potential sources of binding site differences

\begin{enumerate}
\item They first calculated a normalized difference score for ChIP-seq which
does background subtraction and normalization to get the 'binding
signal'
\item Then they analyzed the variation of normdiff across all experiments
to identify cis- and trans- factors contributing to binding site variations

\begin{enumerate}
\item They found that cis-factors, which are DNA changes that were identified
using genetic markers and alignments, were the main source of binding
variations across genomes\\
\includegraphics[scale=0.6]{consensus3}

\begin{enumerate}
\item Quantitatively, 85\% of variable binding regions in a genetic marker
regression laid cis to a genetic marker, and the effect size was greater
for cis-factors than trans-factors
\item Note: see methods/approach for how the marker regression is important
for our goals
\end{enumerate}
\item Zheng et al. also found trans-factors, defined by genetic markers
far away from the binding sites, that were associated with binding
variations (AMN1, FLO8)
\end{enumerate}
\end{enumerate}
\item One of the main results tools from Zheng et al. for finding binding
site variation is the normalized difference (NormDiff) score

\begin{enumerate}
\item NormDiff uses a simple random
\end{enumerate}
\end{enumerate}

\section{Method and approach}

Our method uses ChIP-seq data from multiple genomes and it leverages
the conservation of DNA sequences to find conserved transcription
factor binding sites. We compared ChIP-seq data from two different
yeast strains, S96 and HS959 (Zheng et al 2010), and demonstrate the
need for our proposed method. We used a normalized difference score
to identify additional conserved binding sites, and propose modifications
to consider multiple genomes concurrently.

Identifying conserved binding sites is important for several reasons

One is that variable regions are of interest for identifying functional
differences. Zhenge et al. identified about 900 variable binding regions
using their method, and they only confirmed two of the markers using 

The normalized difference (NormDiff) score is a useful statistic for
comparing ChIP-seq data that uses background subtraction and normalization.
NormDiff uses a simple random model for ChIP-seq and control is defined
as: A\textasciitilde{}Poisson(f+g), B\textasciitilde{}Poisson(cg)
resp. Here, f represents the binding signal, and g represents the
background signal, and c is a scaling factor between A,B Then NormDiff
is defined for each genome position xi as Z(xi)=(A(xi)-B(xi)/c)/\textgreek{sv}
The scaling factor c is estimated as the median ratio of A/B, and
the variance \textgreek{sv} is estimated as \textsurd{}A+B/c\texttwosuperior{}
Then the background distribution Z \textasciitilde{}Normal(0,1), and
deviations from the norm are caused by the binding signal. 

We approached the problem using the normalized difference score. Zheng
et al use the normalized difference score primarily to find binding
site variability
\begin{enumerate}
\item Marker regression is used here to analyze variable binding traits
as QTL, however, analyzing binding site alignments and binding site
conservation is considered by Odom, Dowell et al to be functional
\item In order to characterize these effects, I have to understand QTL mapping
(somewhat difficult) and also develop the algorithm for scoring alignment
of binding sites, and whether these effects in yeast can be characterized
similar to Odom, Dowell et al.\end{enumerate}

\end{document}
