%% LyX 2.0.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english]{article}
\usepackage{mathptmx}
\usepackage{helvet}
\usepackage{courier}
\renewcommand{\familydefault}{\rmdefault}
\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.54cm,bmargin=2.54cm,lmargin=2.54cm,rmargin=2.54cm}
\usepackage{graphicx}
\usepackage{babel}
\begin{document}

\title{First semester final report}


\author{Colin Diesh}


\date{Senior Thesis Project, Fall 2012}
\maketitle
\begin{abstract}
Transcription factors are proteins which recognize short DNA sequences
and they can regulate gene expression by binding to DNA. Using chromatin
immunoprecipitation and high throughput sequencing (ChIP-seq) we can
identify transcription factor binding sites by finding peaks in the
data: places where there is are many DNA reads from immunoprecipitation.
Comparing ChIP-seq data from multiple experiments presents many logistical
challenges -- different ChIP-seq experiments have may have different
signal/noise ratios and using control experiments and normalization
is important. Comparing transcription factor binding sites from different
genomes also presents challenges because binding sites can vary significantly
due to modifications of DNA binding sites, etc.

Previous methods to identify conserved transcription factor binding
sites analyze different genomes separately, then look for binding
sites in conserved and aligned locations {[}1{]}. We use a model for
comparing transcription factor binding from multiple genomes by analyzing
a normalized difference score that uses information from multiple
ChIP-seq experiments. We propose that this method can help find conserved
binding sites across multiple experiments. We evaluated several existing
techniques for analyzing ChIP-seq data analysis and propose our method
for comparing of ChIP-seq data.
\end{abstract}

\section{Background}


\subsection{ChIP-seq basics}

ChIP-seq is a high throughput sequencing method for chromatin immunoprecipitation
(ChIP) for analyzing protein-DNA interactions. ChIP-seq protocols
have been developed to analyze transcription factors and histone modifications
as well as DNA methylation {[}citation{]}. Other ChIP protocols also
use oligonucleotide arrays for tiling the genome (ChIP-chip). ChIP-seq
does not require construction of tiling arrays and so, instead, the
analysis of high throughput sequence data is the main focus.\\



\subsection{Odom, Dowell et al. - Tissue specific transcriptional regulation
has diverged significantly between human and mouse}
\begin{enumerate}
\item Odom, Dowell et al. used ChIP-chip tiling arrays to analyze binding
site differences between human and mouse. 
\item They found orthologous genes in human and mouse and created tiling
arrays from the gene promoter regions
\item They proposed two approaches for data anlysis

\begin{enumerate}
\item - Gene centric approach for ChIP

\begin{enumerate}
\item Classify binding as true/false for each gene (tiling array as a whole)
\item Similar to Zheng et al. by identifying target genes 
\end{enumerate}
\item - Peak centric approach for analysis

\begin{enumerate}
\item Classify peaks within tiling array
\item Find 'peaks' using ChIP-chip and compare aligned sites
\item Classify peaks as conserved, turnover, gain/loss, or unaligned\\
\includegraphics[scale=0.6]{peak_classification}
\item Odom, Dowell results Fig 2c. (Shared binding site patterns)

\begin{enumerate}
\item (1) Alignment in humans/mouse exists and binding sites are aligned
\textasciitilde{}30\%
\item (2) Alignment in human/mouse exists but binding site are not aligned
\textasciitilde{}30\%
\item (3) No alignment in human/mouse exists but binding site still shared
\textasciitilde{}30\%
\end{enumerate}
\item Conclusion: many shared binding sites are not aligned (60\%+) 
\end{enumerate}
\end{enumerate}
\item Questions about Odom, Dowell

\begin{enumerate}
\item How can we use their results to answer questions about ChIP-seq where
sequence comparison is important for ChIP-seq sequence similarity?
\end{enumerate}
\end{enumerate}

\subsection{Zheng et al. \textendash{} Genetic analysis of transcription factor
binding variation in yeast}
\begin{enumerate}
\item Zheng et al. analyzed binding sites from many yeast ChIP-experiments
using and found potential sources of binding site differences

\begin{enumerate}
\item They first calculated a normalized difference score for ChIP-seq which
does background subtraction and normalization for the ChIP-seq data
\item Then second they analyzed the variation of normdiff scores across
all experiments to identify cis- and trans- factors contributing to
binding site variations

\begin{enumerate}
\item They found that cis-factors, which are DNA changes identified using
genetic markers and alignments, were the main source of binding variations
across genomes
\item They also found trans-factors, defined by genetic markers far away
from the binding sites, that were associated with binding variations
(AMN1, FLO8)
\end{enumerate}
\item They also analyzed linkage and transgressive inheritance
\end{enumerate}
\item One of the main results tools from Zheng et al. for finding binding
site variation is the normalized difference (NormDiff) score

\begin{enumerate}
\item NormDiff uses a simple random
\end{enumerate}
\end{enumerate}

\section{Method and approach}

We approached the problem using the normalized difference score
\end{document}
