%% LyX 2.0.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english]{article}
\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.54cm,bmargin=2.54cm,lmargin=2.54cm,rmargin=2.54cm}
\usepackage{babel}
\begin{document}

\title{First semester final report}


\author{Colin Diesh}


\date{Senior Thesis Project, Fall 2012}
\maketitle
\begin{abstract}
Transcription factors are proteins which recognize short DNA sequences
and bind to the DNA which can affect transcription both positively
and negatively. Using chromatin immunoprecipitation and high throughput
sequencing (ChIP-seq) we can identify transcription factor binding
sites by finding peaks in the data, or where there is an excess of
DNA reads from immunoprecipitation. Comparing ChIP-seq data from multiple
experiments presents many logistic challenges -- different ChIP-seq
experiments have may have different signal/noise ratios and using
control experiments and normalization is important. In order to analyze
data from differentComparing data from different genomes 

We investigated methods for comparing transcription factor binding
from multiple Chip-Seq experiments. Previous methods to identify conserved
transcription factor binding sites from different experiments analyze
different genomes separately, then look for binding sites in conserved
locations. Transcription factor binding is We investigated methods
for comparing transcription factor binding from multiple Chip-Seq
experiments. We evaluated several existing techniques for ChIP-seq
data analysis, and we propose a method for finding ChIP-seq data similarity
using a normalized difference score to leverage conservation between
related experiments.\end{abstract}

\end{document}
