%% LyX 2.0.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english]{article}
\usepackage{mathptmx}
\usepackage{helvet}
\usepackage{courier}
\renewcommand{\familydefault}{\rmdefault}
\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.54cm,bmargin=2.54cm,lmargin=2.54cm,rmargin=2.54cm}
\usepackage{graphicx}
\usepackage{babel}
\begin{document}

\title{First semester final report}


\author{Colin Diesh}


\date{Senior Thesis Project, Fall 2012}
\maketitle
\begin{abstract}
Transcription factors are proteins which recognize and bind to short
DNA sequences, and they can play an important role in regulating DNA
transcription in a gene regulatory network. Comparing transcription
factor binding sites from different related species can show important
functional similarities and differences, however transcription factor
binding sites have also been shown to vary significantly across related
species and strains. Zheng et al characterized transcription factor
binding variations by using a normalized difference score to find
'variable binding regions' across different experiments. We use the
normalized difference score to find conserved binding sites. We also
address issues of DNA alignment from different genomes in order to
classify binding sites similar to Odom, Dowell et al 2007. 
\end{abstract}

\section{Background}

Chromatin immunoprecipitation is a methodology for analyzing protein-DNA
interactions. Antibodies that bind protein-DNA complexes have been
developed to do ChIP with transcription factors, histones, and DNA
methylation (meDIPSeq). Using chromatin immunoprecipitation and high
throughput sequencing (ChIP-seq) or oligonucleotide arrays for tiling
the genome (ChIP-chip) we can identify transcription factor binding
sites. by finding peaks in the data we can analyze transcription factor
binding. Here we will look at some background papers in the field\\



\subsection{Tissue specific transcriptional regulation has diverged significantly
between human and mouse (Odom, Dowell et al.)}
\begin{enumerate}
\item Odom, Dowell et al. used ChIP-chip tiling arrays to analyze binding
site differences between human and mouse. 
\item They found orthologous genes in human and mouse and created tiling
arrays from the gene promoter regions
\item They proposed two approaches for data anlysis

\begin{enumerate}
\item - Gene centric approach for ChIP

\begin{enumerate}
\item Classify binding as true/false for each gene (tiling array as a whole)
\item Similar to Zheng et al. by identifying target genes 
\end{enumerate}
\item - Peak centric approach for analysis

\begin{enumerate}
\item Classify peaks within tiling array
\item Find 'peaks' using ChIP-chip and compare aligned sites
\item Classify peaks as conserved, turnover, gain/loss, or unaligned\\
\includegraphics[scale=0.6]{peak_classification}
\item Odom, Dowell results Fig 2c. (Shared binding site patterns)

\begin{enumerate}
\item (1) Alignment in humans/mouse exists and binding sites are aligned
\textasciitilde{}30\%
\item (2) Alignment in human/mouse exists but binding site are not aligned
\textasciitilde{}30\%
\item (3) No alignment in human/mouse exists but binding site still shared
\textasciitilde{}30\%
\end{enumerate}
\item Conclusion: many shared binding sites are not aligned (60\%+) 
\end{enumerate}
\end{enumerate}
\item Questions about Odom, Dowell

\begin{enumerate}
\item How can we use their results to answer questions about ChIP-seq where
sequence comparison is important for ChIP-seq sequence similarity?
\end{enumerate}
\end{enumerate}

\subsection{Zheng et al. \textendash{} Genetic analysis of transcription factor
binding variation in yeast}
\begin{enumerate}
\item Zheng et al. analyzed binding sites from many yeast ChIP-experiments
using and found potential sources of binding site differences

\begin{enumerate}
\item They first calculated a normalized difference score for ChIP-seq which
does background subtraction and normalization to get the 'binding
signal'
\item Then they analyzed the variation of normdiff across all experiments
to identify cis- and trans- factors contributing to binding site variations

\begin{enumerate}
\item They found that cis-factors, which are DNA changes that were identified
using genetic markers and alignments, were the main source of binding
variations across genomes\\
\includegraphics[scale=0.6]{consensus3}

\begin{enumerate}
\item They used a 'single marker regression' to associate variable binding
regions with genetic markers and found that 85\% of variable binding
regions in a genetic marker regression laid cis to a genetic marker,
and the effect size was greater for cis-factors than trans-factors
\item Note: see methods/approach for how the marker regression is important
for our goals
\end{enumerate}
\item Zheng et al. also found trans-factors, defined by genetic markers
far away from the binding sites, that were associated with binding
variations (AMN1, FLO8)
\end{enumerate}
\end{enumerate}
\item One of the main results tools from Zheng et al. is the use of NormDiff

\begin{enumerate}
\item NormDiff is used to find 'variable binding regions', and these are
linked with genetic markers using 'single marker regression'

\begin{enumerate}
\item This technique is adopted from QTL analysis, which I don't fully understand,
but as far as i can tell, it does not address which sites are conserved,
or which sites are aligned, but relies on genetic markers for this.
\item Can we address these issues systematically to find--

\begin{enumerate}
\item Conserved binding sites?
\item Aligned binding sites?
\end{enumerate}
\end{enumerate}
\end{enumerate}
\end{enumerate}

\section{Method and approach}

We use ChIP-seq data gathered from different genomes and the alignment
of DNA sequences to classify conserved transcription factor binding
sites similar to Odom, Dowell et al 2007. We compared ChIP-seq data
from two different yeast strains, S288c and YJM789, (Zheng et al 2010),
and . We used a normalized difference score to identify additional
conserved binding sites, and propose modifications to consider multiple
genomes concurrently.


\subsection{Normalized difference scores}

The normalized difference (NormDiff) score is a useful statistic for
comparing ChIP-seq data that uses background subtraction and normalization.
NormDiff uses a simple random model for ChIP-seq ($A$) and control
($B$) is defined as

\begin{eqnarray*}
A & \sim & Poisson(f+g)\\
B & \sim & Poisson(cg)
\end{eqnarray*}


Where- 

$f$ represents the binding signal

$g$ represents the background signal

$c$ is a scaling factor between $A$ and $B$\\


Then, NormDiff is defined for each genome position $x_{i}$ as 

\[
Z(x_{i})=\frac{A(x_{i})-B(x_{i})/c}{\sigma}
\]
 

Then the scaling factor $c$ is estimated as the median ratio of $A/B$
and the variance $\sigma$ is estimated from the maximum variance
of $\sqrt{A+B/c^{2}}$ in a local window. Then, the background score
is where $Z~Normal(0,1)$ and deviations from the norm are caused
by the strong ChIP-seq binding signals. 


\section{Problem statement}
\begin{itemize}
\item Zheng et al use the normalized difference score primarily to find
binding site variability. This approach is useful in finding quantitative
differences in binding using a QTL analysis. 
\item We wanted to perform a more detailed analysis of their approach for
applying QTL analysis to transcription factor binding using NormDiff. 
\item We want to characterize conserved binding sites using peak centric
and alignments Odom, Dowell et al's. What will this get us? It will
classify the 900+ 'variable binding sites' that Zheng et al found
for example. 
\item The only 'classification' that is made is whether cis-factors or trans-factors
are associated via QTL and their effect\\
\\
\includegraphics[scale=0.6]{midnight}\end{itemize}
\begin{enumerate}
\item They use marker regression is to analyze variable binding traits as
cis- and trans- factors as QTL

\begin{enumerate}
\item In order to characterize marker regression, we have to understand
QTL mapping
\item Then we have to develop the algorithm for scoring alignment of binding
sites similar to Odom, Dowell et al.\end{enumerate}
\end{enumerate}

\end{document}
