%% LyX 2.0.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english]{article}
\usepackage{mathptmx}
\usepackage{helvet}
\usepackage{courier}
\renewcommand{\familydefault}{\rmdefault}
\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.54cm,bmargin=2.54cm,lmargin=2.54cm,rmargin=2.54cm}
\usepackage{graphicx}

\makeatletter

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% LyX specific LaTeX commands.
%% Because html converters don't know tabularnewline
\providecommand{\tabularnewline}{\\}

\makeatother

\usepackage{babel}
\begin{document}

\title{Comparison and analysis of transcription factor binding sites}


\author{Colin Diesh}


\date{Senior Thesis Project, Fall 2012\\
$\,$\\
First semester final report - Outline}
\maketitle
\begin{abstract}
Transcription factors are proteins which recognize and bind to short
DNA sequences, and they can play an important role in regulating DNA
transcription in a gene regulatory network. Comparing transcription
factor binding sites from different related species can show functional
differences, however even functionally conserved binding sites have
been shown to vary significantly. Zheng et al. characterized transcription
factor binding variations by using a normalized difference score to
find 'variable binding regions' across different experiments. We use
statistical comparison of normalized difference score to find conserved
binding sites and we also use the alignment from different genomes
in order to classify binding sites. 
\end{abstract}

\section{Background}

Protein-DNA interactions play an important role in gene regulation,
and chromatin immunoprecipitation (ChIP) is a methodology for analyzing
these interactions. Antibodies that bind to transcription factors
or histone modifiers are used to immunoprecipitate the protein-DNA
complexes. Using chromatin immunoprecipitation and high throughput
sequencing (ChIP-seq) or oligonucleotide tiling arrays (ChIP-chip)
we can identify transcription factor binding sites. We search for
peaks in the data that correspond to an enrichment of ChIP in the
sequence, which are called binding sites. Comparing ChIP data from
multiple experiments is important to identify functional differences
Here we will look at some background papers in the field\\



\subsection{Tissue specific transcriptional regulation has diverged significantly
between human and mouse (Odom, Dowell et al.)}
\begin{enumerate}
\item Odom, Dowell et al. used ChIP-chip tiling arrays to analyze binding
site differences in human and mouse hapetocytes (liver tissue). 
\item They used orthologous genes in human and mouse, and they created tiling
arrays from the gene promoter regions
\item They proposed two approaches for data anlysis

\begin{enumerate}
\item - Gene centric approach for ChIP

\begin{enumerate}
\item Classify binding as true/false for each gene (tiling array as a whole)
\item Similar to Zheng et al. by identifying target genes 
\end{enumerate}
\item - Peak centric approach for analysis

\begin{enumerate}
\item Classify peaks within tiling array
\item Find 'peaks' using ChIP-chip and compare aligned sites
\item Classify peaks as conserved, turnover, gain/loss, or unaligned\\
\includegraphics[scale=0.6]{peak_classification}\\
Figure 1. Peak classifications from Odom, Dowell et al.\\

\item Odom, Dowell results Fig 2c. (Shared binding site patterns)

\begin{enumerate}
\item (1) Alignment in humans/mouse exists and binding sites are aligned
\textasciitilde{}30\%
\item (2) Alignment in human/mouse exists but binding site are not aligned
\textasciitilde{}30\%
\item (3) No alignment in human/mouse exists but binding site still shared
\textasciitilde{}30\%
\end{enumerate}
\item Conclusion: many shared binding sites are not aligned (60\%+) \\
\\
\begin{tabular}{|c|c|c|c|c|}
\hline 
 & FOXA2  & HNF1A & HNF4A & HNF6\tabularnewline
\hline 
\hline 
Binding sites aligns & 34\% & 41\%  & 23\% & 30\%\tabularnewline
\hline 
Binding sites do not align & 39\% & 31\% & 38\%  & 41\%\tabularnewline
\hline 
No shared alignment & 28\% & 28\% & 38\%  & 30\%\tabularnewline
\hline 
\end{tabular}
\end{enumerate}
\end{enumerate}
\item Questions about Odom, Dowell

\begin{enumerate}
\item How can we use their results to answer questions about ChIP-seq where
sequence comparison is important for ChIP-seq sequence similarity?
\end{enumerate}
\end{enumerate}

\subsection{Zheng et al. \textendash{} Genetic analysis of transcription factor
binding variation in yeast}
\begin{enumerate}
\item Zheng et al. analyzed binding sites from many yeast ChIP-experiments
using and found potential sources of binding site differences

\begin{enumerate}
\item They first calculated a normalized difference score for ChIP-seq which
does background subtraction and normalization to get the 'binding
signal'
\item Then they analyzed the variation of normdiff across all experiments
to identify cis- and trans- factors contributing to binding site variations

\begin{enumerate}
\item They found that cis-factors, which are DNA changes that were identified
using genetic markers and alignments, were the main source of binding
variations across genomes\\
\\
\includegraphics[scale=0.6]{consensus3}\\
Figure 2. cis-factor correlated with variable binding is a modified
transcription factor binding motif

\begin{enumerate}
\item They used a 'single marker regression' to associate variable binding
regions with genetic markers and found that 85\% of variable binding
regions in a genetic marker regression laid cis to a genetic marker,
and the effect size was greater for cis-factors than trans-factors
\item Note: see methods/approach for how the marker regression is important
for our goals
\end{enumerate}
\item Zheng et al. also found trans-factors, defined by genetic markers
far away from the binding sites, that were associated with binding
variations (AMN1, FLO8)
\end{enumerate}
\end{enumerate}
\item One of the main results tools from Zheng et al. is the use of NormDiff

\begin{enumerate}
\item NormDiff is used to find 'variable binding regions' by finding regions
where normdiff scores have a high degree of variability across experiments

\begin{enumerate}
\item Then, variable binding regions are linked with genetic markers using
'single marker regression', a technique used to analyze quantitative
trait loci (QTL)
\item The genetic markers indicate SNP and indel, but this method does not
explicitely address the alignment of binding sites.
\item Furthermore, they found many variable binding regions, on the order
of 900, so classifying them using an alignment would be useful
\item Can we address these issues systematically to find--

\begin{enumerate}
\item Conserved binding sites?
\item Aligned binding sites?
\end{enumerate}
\end{enumerate}
\end{enumerate}
\end{enumerate}

\section{Method and approach}

We use ChIP-seq data gathered from two seperate yeast strains (S288c
and YJM789) and then we used alignment of DNA sequences to classify
conserved transcription factor binding sites similar to Odom, Dowell
et al 2007. We used a normalized difference score that was defined
by Zheng et al. to identify conserved binding sites, and propose modifications
to consider multiple genomes concurrently.


\subsection{Normalized difference scores}

The normalized difference (NormDiff) score is a useful statistic for
comparing ChIP-seq data that uses background subtraction and normalization
to obtain the ChIP-seq binding signal. NormDiff uses a simple random
model for ChIP-seq ($A$) and control ($B$) which is defined as

\begin{eqnarray*}
A & \sim & Poisson(f+g)\\
B & \sim & Poisson(cg)
\end{eqnarray*}


Where- 

$f$ represents the binding signal

$g$ represents the background noise

$c$ is a scaling factor between $A$ and $B$\\


Then, NormDiff is defined for each genome position $x_{i}$ as 

\[
Z(x_{i})=\frac{A(x_{i})-B(x_{i})/c}{\sigma}
\]
 

Then the scaling factor $c$ is estimated as the median ratio of $A/B$
and the variance $\sigma$ is estimated from the maximum variance
of $\sqrt{A+B/c^{2}}$ in a local window. Then, the background score
is where $Z~Normal(0,1)$ and deviations from the norm are caused
by the strong ChIP-seq binding signals. 


\section{Problem statement}
\begin{itemize}
\item Zheng et al use the normalized difference score primarily to find
binding site variations. This approach is useful in finding quantitative
binding differences, and they use their technique for QTL mapping. 
\item We wanted to perform a more detailed analysis of their approach by
evaluating the alignment of binding sites transcription factor binding
using NormDiff. 
\item We want to classify binding sites using their alignment as being conserved,
gain/loss, turnover, or even unaligned

\begin{itemize}
\item This can help in analyzing the many (900+) 'variable binding sites'
that Zheng et al found in their analysis.
\item The classification made by Zheng et al is linking variable binding
with genetic markers
\item This is a technique from QTL analysis to associate cis-factors or
trans-factors with their effects (binding variation)\\
\\
\includegraphics[scale=0.6]{midnight}\\
Figure 3. The cis-factors explain a larger proportion of variable
binding than trans-factors
\end{itemize}
\end{itemize}

\section{Previous work and application}

We were interested in finding conserved binding sites by looking for
evidence of binding sites even where binding sites were not called
by MACS. We used an algorithm that reanalyzes binding sites using
the normalized difference score.


\subsection{Experimental approach}
\begin{itemize}
\item We identified a set of high confidence peaks in ChIP-seq data using
MACS 
\item Then we used a sliding window to examine the maximum average NormDiff
peak score overlapping the peaks and the syntenic regions of each
ChIP-seq experiment 
\item Based on the background distribution $Z\sim Normal(0,1)$ we used
the maximum average NormDiff score of our scanning window to calculate
the probability of observing a peak as $P(z\leq Z)$ 
\item Conserved binding sites were identified as samples where NormDiff
scores were highly different from the background distribution. \\
\\
\includegraphics[scale=0.3]{untitled1}\\
Figure 4. The HS959 normdiff score of shared peaks with S96 are higher
than the HS959 normdiff scores for peaks unique to S96. However, there
is significant overlap between the normdiff scores from each
\end{itemize}

\subsection{Results}
\begin{itemize}
\item We found 42 new binding sites in HS959 with P<0.05 that were conserved
in S96 that were not identified by MACS. Also 4 of these binding sites
were conserved with P<0.01
\item We found 76 binding sites in S96 with P<0.05 that were conserved in
HS959, and 13 of these binding sites were conserved with P<0.01 
\item To evaluate the significance of these findings, we evaluated peaks
in S96 using normalized difference scores and identified 98\% (885/897)
of the binding sites that MACS identifies at P<0.05 confidence level,
and 60\% (563/897) binding sites at a P<0.01. 
\item In HS959 we identified 85\% (729/824) of the same peaks that MACS
identifies at a 0.05 confidence level and 46\% (365/824) peaks at
P<0.01\\
\includegraphics[scale=0.3]{untitled3}\\
Figure 5. The maximum average NormDiff score is used to identify additional
binding sites that are conserved in each experiment. \end{itemize}

\end{document}
