---
title       : Comparison and analysis of ChIP-seq experiments
subtitle    : 
author      : Colin Diesh
job         : 
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js  # {highlight.js, prettify, highlight}
hitheme     : tomorrow      # 
widgets     : []            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
---

## Outline
# Overview of the presentation

- Background
  - What is ChIP-seq?
  - Overview of problem
- Comparing ChIP-seq data
  - Review of existing methods
  - My Approach
    - Finding differential binding sites
- Results
  - Evaluate method
  - Compare with other methods
- Conclusion


--- 

## Background

# What is ChIP-seq?


- ChIP-seq = High-throughput sequencing + chromatin immunoprecipitation
- Produces millions of short DNA reads that are enriched for DNA-protein interactions


![chipseq-overview](assets/img/image1.png)
Huber et al, 2006

--- 


## Background

# What does ChIP-seq look like?

- It's noisey!
- The distribution of the ChIP-seq reads is noisy and depends on the protocol and sequencing depth

```{r echo=FALSE,message=FALSE,fig.width=12,fig.height=5,warning=FALSE,results='hide'}
require(ProjectTemplate)
od<-setwd('..')
load.project()

plot(macswiggle[[1]]$control$chr01$V1,macswiggle[[1]]$control$chr01$V2,'l',col=1,xlab="Chr01 position",ylab="Number of ChIP-seq tags overlapping position",ylim=c(0,80))
lines(macswiggle[[1]]$treat$chr01$V1,macswiggle[[1]]$treat$chr01$V2,col="#AA2222AA")
title('Distribution of ChIP-seq and control reads (Chr01)')
setwd(od)
```
--- 

## Background

# Analyzing the data--finding peaks in the data

- Use knowledge about data to build peak model
- Peak model uses distribution of paired-end reads to estimate the size of the DNA fragments
  - Sequencer reads are ~20-50bp
  - Fragment size is between ~100-300bp


![chipseq-fragments](assets/img/image2.png)

Wilbanks et al, 2010


---


## Overview 
# ChIP-seq applications

- ChIP-seq can be used to learn about
  - transcription factor binding sites
  - histone modifications
  - regulatory motifs
- Important to compare datasets from experiments
- Find effects of binding sites regulation on gene expression

![regulatory-network](assets/img/image4.gif)
From http://genome.tbdb.org/annotation/genome/tbdb/RegulatoryNetwork.html

---

## Overview
# Comparing ChIP-seq data


- Binding sites can move, disappear, and change
- Comparing related genomes can provide some insight

![odom-compare](assets/img/image8.png) 


Odom, Dowell et al. 2007

---


