%% LyX 2.0.3 created this file.  For more info, see http://www.lyx.org/.
%% Do not edit unless you really know what you are doing.
\documentclass[english]{article}
\usepackage{mathptmx}
\usepackage{helvet}
\usepackage{courier}
\usepackage[T1]{fontenc}
\usepackage[latin9]{inputenc}
\usepackage{geometry}
\geometry{verbose,tmargin=2.54cm,bmargin=2.54cm,lmargin=2.54cm,rmargin=2.54cm}
\usepackage{graphicx}
\usepackage{babel}
\begin{document}

\title{Comparison and analysis of ChIP-seq experiments}


\author{Colin Diesh}


\date{Senior Thesis Project, Fall 2012\\
$\,$\\
First semester final report - Outline}
\maketitle
\begin{abstract}
DNA-protein interactions are essential for gene regulation, and these
interactions can be detected by using chromatin immunoprecipitation
and high throughput sequencing (ChIP-seq). In particular, transcription
factor binding sites are identified as a significant enrichment of
the ChIP-seq data in particular DNA regions. In order to compare data
from different experiments, we want to normalize the datasets to find
shared significant binding sites. We use the normalized difference
score described by Zheng et al (2010) to find evidence for shared
binding sites from different ChIP-seq experiments.
\end{abstract}

\section{Introduction}

The genes in our DNA are regulated by a variety of epigenetic factors.
Transcription factors in particular are proteins that can affect DNA
transcription by binding to the DNA. Transcription factors are characterized
by having a specific DNA binding domain which is a short DNA sequence
which they are attracted to. These transcription factor binding sites
play a important role in regulating DNA transcription. 

ChIP-seq is a technique used for experimentally finding transcription
factor binding sites. ChIP-seq uses chromatin immunoprecipitation
to enrich for DNA fragments with transcription factors attached followed
by DNA sequencing of the fragments. Then transcription factor binding
sites appear as enriched \textquoteleft{}peaks\textquoteright{} in
the ChIP-seq data. The significance level for finding peaks is based
on a p-value, but in order to compare data from different experiments,
the data sets are normalized to find shared significant binding sites. 

We used the normalized difference (NormDiff) score described by Zheng
et al (2010) to get the background subtracted and normalized ChIP-seq
binding signal. We use the binding sites identified from MACS and
we compared the NormDiff score from the syntenic binding sites from
multiple experiments to find evidence for shared binding sites. 


\section{Background}

In 2010, Zheng et al. analyzed the differences in transcription factor
binding sites from two different strains of yeast using ChIP-seq.
They used ChIP-seq experiments to find transcription factor binding
sites of Ste12, a transcription factor which is important in yeast
mating. For analyzing binding sites they used a peak finding software
called MACS \textendash{} Model based analysis of ChIP-seq. Then they
used a normalized difference score to see the binding signal and they
analyzed the variation of the signal across each experiment. By genotyping
each strain they analyzed, they were able to find the genetic factors
that underlied the variations in transcription factor binding sites
also.


\subsection{Model based analysis of ChIP-seq (Zhang et al. 2007)}

To understand how ChIP-seq analysis works we studied the MACS, a popular
open source ChIP-seq analysis tool, to find out details about the
processing of ChIP-seq data. The published description of the program
is quite thorough, but we found additional info that helped us understand
the details of ChIP-seq processing. MACS is a command line program
that has a number of configurability options and understanding the
processing that MACS does is important for understanding these options. 

MACS takes user input from the command line and loads ChIP-seq data
into a Python hash data structure called a dictionary. The dictionary
associates the chromosome of each ChIP-seq tag as the hash-key, and
append the position and orientation of the tags into a list. MACS
does not use store or use any other the details of the reads, such
as the sequence of the number of mismatches{*}.

The MACS calculates a \textquotedblleft{}shift model\textquotedblright{}
to find the offset between +/- strands. This step is necessary to
compensate for the fact that only part of the 5\textquoteright{} ends
of the ChIP-seq fragments are sequenced. The fragments are assumed
to be distributed randomly on each strand, so MACS uses a \textquotedblleft{}naïve\textquotedblright{}
peak finder to find pairs of peaks on each strand, and calculates
the median distance between each pair to estimate the DNA fragment
size. Then MACS extends each read towards the 3\textquoteright{} end
to represent the true length of the ChIP-seq fragment.

Then MACS does peak finding for all shifted reads by using Poisson
distribution model. The rate parameter of the Poisson is calculated
from the expected number of reads in our scan window given the total
number of reads and the whole genome size. Then peaks are identified
as the Poisson CDF likelihood of receiving a large number of reads
in our scan window given the Poisson rate of the background. The Poisson
rate parameter is also estimated for local windows surrounding the
scanning window for 5k and 10k windows to local capture variations
in the data. 

\includegraphics[scale=0.5]{Poisson_cdf}

Figure 1. The Poisson cumulative distribution function (CDF) outputs
a probability for observing a given number of events.\\


{*}It is possible that analysis could reveal that mismatch alignments
somehow correlate with ChIP-seq data features


\subsection{Finding conserved transcription factor binding sites}

Finding conserved binding sites from different experiments is important
for measuring data integrity, reproducibility, and for finding biological
invariants. However, ChIP-seq experiments are subject to many variables
such as noise, bias, and differences in sequencing depth (Shao et
al. 2012). These variables can make comparing different ChIP-seq experiments
difficult, so it is important to find a good model for comparing the
datasets.

One method for comparing data is to control look at the empirical
false discovery rate given by MACS software. This method can help
identify a shared significant binding level, since if we use a low
significance threshold to find shared binding sites, we may get too
many false positives. In this situation, we can use the empirical
false discovery rate (FDR) calculated by MACS to guide our analysis.
The FDR is calculated for each peak as the ratio of the number of
peaks that MACS finds in the control dataset vs the number of peaks
found in the ChIP-seq dataset for the p-value that is measured at
that peak. Using a FDR less than 1\% is recommended for peak calling
and conservation analysis (Bardet, 2012). Controlling for FDR from
MACS algorithm helps find shared significant peaks without setting
a strict p-value threshold for peak finding.

Another method for identifying conserved binding sites is \textquotedblleft{}MANorm:
a robust model for quantitative comparison of ChIP-seq datasets\textquotedblright{}
(Shao et al, 2012). MANorm uses the total number of reads that are
found in an initial finding of the shared binding sites from two ChIP-seq
datasets. Assuming the read count from shared binding sites should
be similar, it uses a linear model to fit these peaks, and then it
extrapolates the model to all reads in order to find additional shared
binding sites. MANorm also uses the binding sites identified in one
experiment as prior evidence for finding shared binding sites by used
a Bayesian Poisson model. 

MANorm effectively solves the problem of using one experiment as prior
information for finding shared binding sites. Nevertheless, we developed
our own model for assessing binding site similarity using the normalized
difference score that is described in our approach. 

\includegraphics[scale=0.4]{\string"MAnorm_ a robust model for quantitative  - Zhen Shao\string".eps}

Figure 2. MANorm workflow for normalizing ChIP-seq data and finding
conserved binding sites


\section{Approach}

In our study we found that MACS found many peaks that were not shared
from ChIP-seq experiments from closely related yeast strains by Zheng
et al. In order to compare ChIP-seq data and find conserved binding
sites, we use a normalized difference (NormDiff) score to calculate
the \textquotedblleft{}background subtracted\textquotedblright{} ChIP-seq
binding signal. We used MACS to find high confidence peaks from both
ChIP-seq datasets, and we then we used the normalized difference score
get evidence for shared binding sites that were not found by MACS. 

To find evidence for the shared binding sites, we compared the NormDiff
scores from binding sites in one experiment with the syntenic regions
of the binding sites in the other experiment. This analysis showed
enriched NormDiff scores in the syntenic regions which indicates enriched
ChIP-seq binding signal (Figure 3). Using the normal distribution,
we calculated shared binding sites from NormDiff scores as a statistical
likelihood that they are shared.

\includegraphics[scale=0.4]{Rplot02_2}\includegraphics[scale=0.4]{Rplot03}

Figure 3. The maximum average NormDiff score of peaks shared binding
sites is colored red but unique binding sites identified in only one
experiment are colored blue/green


\subsection{Results}

We found 42 new binding sites in HS959 with that were conserved in
S96 that were not identified by MACS with P<0.05, and 4 of binding
sites that were conserved with P<0.01. We also found 76 binding sites
in S96 with P<0.05 that were conserved in HS959, and 13 of these binding
sites were conserved with P<0.01 

To evaluate the significance of these findings, we evaluated all binding
sites from S96 and identified 98\% (885/897) using our meethod at
P<0.05, and 60\% (563/897) binding sites at a P<0.01. In HS959 we
identified 85\% (729/824) of the same binding sites at P<0.05 and
46\% (365/824) at P<0.01\\
\includegraphics[scale=0.4]{Rplot23}\includegraphics[scale=0.4]{Rplot25_2}\\
Figure 5. The maximum average NormDiff score is used to identify additional
binding sites that are conserved in each experiment. 


\section{Discussion}

Using our method there appeared to be evidence for conserved binding
sites that were not identified by other peak finding algorithms. We
see a significant overlap of NormDiff scores for many binding sites,
which represents strong binding signals, even for binding sites that
were were not identified as shared by MACS. We call this area of overlap
the \textquotedblleft{}twilight zone\textquotedblright{} which represents
potentially shared binding sites at a level not identified by peak
finders at high confidence levels.

\includegraphics[scale=0.3]{untitled1}\includegraphics[scale=0.3]{untitled2}

Figure 5. The twilight zone in yellow shows where the normalized difference
scores from each experiment overlap significantly \\


\includegraphics[scale=0.3]{\string"hgt_genome_249b_9588c0 poster\string".eps}

Figure 6. A shared binding sites that we identified using our algorithm
has the above distribution.\\


Our work could be extended to use more sophisticated statistical models
for NormDiff too. For example, we considered modifying NormDiff scores
to use data from multiple experiments in order to find conserved binding
sites. We also discuss some of the drawbacks of our method, and the
other applications of NormDiff. 


\subsection{Modified normdiff score}

We propose using a modified NormDiff score that uses data from multiple
experiments. A NormDiff score that adds data from two experiments,
A1,B1,A2, and B2 can be defined as \\
\[
Z_{add}(x_{i})=\frac{(A_{1}-B_{1}/c_{1})+(A_{2}-B_{2}/c_{2})}{\sigma}
\]


Then the scaling factors $c_{1}$and $c_{2}$ are estimated from data,
and the variance is $\sigma=\sqrt{A_{1}+B_{1}/c_{1}^{2}+A_{2}+B_{2}/c{}_{2}^{2}}$ 

A NormDiff score that subtracts data from two experiments could also
be defined as
\[
Z_{sub}(x_{i})=\frac{(A_{1}-B_{1}/c_{1})-(A_{2}-B_{2}/c_{2})}{\sigma}
\]


The variance for $Z_{sub}$ would be the same as the variance for
$Z_{add}$. It is possible that adding and subtracting NormDiff scores
might be able to help in identifying conserved and differential binding.
For example, if the subtractive score is small but the additive score
is large, then the binding site is likely to be conserved in both
experiments.\\


\includegraphics[scale=0.4]{Rplot28}

Figure 6. $Z_{add}$ vs. $Z_{sub}$ for S96 and HS959. This plot is
similar to a MA plot because log product and log ratio are similar
to adding and subtracting the scores. The unique and shared binding
sites are colored

\includegraphics[scale=0.4]{Rplot05}

Figure 7. MA-plot of shared and unique peaks from S96 and HS959. Unique
peaks to S96 colored green, unique peaks to HS959 colored red, shared
peaks from both colored blue.

.


\subsection{Drawbacks of our approach}

Our approach tries to find evidence for conserved binding sites by
analyzing data using normalized difference scores. Our approach gives
some evidence of peaks with high confidence P-values, such as our
results of finding new peaks with P<0.01. However, these findings
have much less confidence than thresholds set by MACS (P<1e-5). Given
that our technique for calculating P-value is similar to MACS, by
finding large deviations from the background distribution, our results
are not very strong. 

Since NormDiff has much less confidence in finding conserved binding
sites, it may be unproductive to use our method. One of the reasons
for having less confidence in P-values is because NormDiff uses control
data for background subtraction. MACS on the other hand uses only
control data to calculate FDR, and only ChIP-seq data to calculate
P-values. Additionally, we use a statistical distribution which is
less characteristic of our data, the normal distribution, and since
our data is not continuous it does not behave well on small datasets
such the NormDiff scanning window. While our method is not productive
for finding conserved binding sites in the way we describe, it can
be used for other applications. 


\subsection{Applications of NormDiff}

NormDiff scores has other interesting applications besides our method.
One example from Zheng et al. is for finding variable binding regions
across many ChIP-seq experiments. This looks at the variability of
the NormDiff score and finds regions of high variability as variable
binding regions. These variable binding regions were correlated with
genetic features using quantitative trait loci and validated using
hierarchial Bayesian clustering algorithms.\\


\includegraphics[scale=0.5]{\string"Figure 1\string".eps}

Figure 2. Hierarchical clustering of variable binding traits identified
using NormDiff associated with genetic markers using QTL (Zheng et
al 2010) \\


Another example of normalized difference scores application is the
\textquotedblleft{}transgressive score\textquotedblright{} which is
used by Zheng et al. to evaluate whether binding sites from yeast
cross strains were inherited from their parents. The transgressive
score analyzes the variance of normdiff scores across all children
compared with the the difference of the parents: 

\[
transgressive=\frac{var(Z)}{Z_{parent1}-Z_{parent2}}
\]


They say that if the transgressive score is low, then the binding
site is ``inherited'' in a Mandelian fashion from one of the parents,
and so in some sense, it was conserved. However, if the transgressive
score is high, then the binding site is considered an extreme phenotype
that is transgressive. Their analysis shows in yeast that some even
lowly expressed peaks might be called transgressive because they are
not conserved with the same strength of the parents' binding sites. 

\includegraphics[scale=0.6]{untitled4}

Figure 8. ``ChIP-Seq signal tracks showing Ste12-binding sites that
segregate in a Mendelian (left) or transgressive (middle and right)
fashion. The colour indicates genotype background in the depicted
regions: red (S96), green (HS959). Asterisks indicate peaks of interest''
(Zheng et al. 2010)


\section{Conclusion}

We found a method for comparing binding sites from different ChIP-seq
experiments by using a normalized difference score. This gave us evidence
for shared binding sites that were not called by other peak finding
algorithms. While it was able to produce some evidence, the confidence
levels were much less than MACS which presented a challenge to the
conclusions we could make. Despite these issues, NormDiff is a powerful
tool for analyzing ChIP-seq data in other ways. NormDiff effectively
models the ChIP-seq binding signal and it can be compared across multiple
ChIP-seq experiments. The NormDiff score can also be analyzed as a
quantitative trait in QTL analysis to give us insight into transcription
factor binding. 


\section{Methods}


\subsection{Normalized difference scores}

The normalized difference (NormDiff) score is a useful statistic for
comparing ChIP-seq data that uses background subtraction and normalization
to obtain the ChIP-seq binding signal. The NormDiff score from Zheng
et al. uses a simple random model for ChIP-seq ($A$) and control
($B$) defined as

\begin{eqnarray*}
A & \sim & Poisson(f+g)\\
B & \sim & Poisson(cg)
\end{eqnarray*}


Where- 

$f$ represents the binding signal

$g$ represents the background noise

$c$ is a scaling factor between $A$ and $B$\\


Then, the NormDiff $Z$ is defined for each genome position $x_{i}$
as 

\[
Z(x_{i})=\frac{A(x_{i})-B(x_{i})/c}{\sigma}
\]
 

Then the scaling factor $c$ is estimated as the median ratio of $A/B$
and the variance $\sigma$ is estimated from the maximum variance
of $\sqrt{A+B/c^{2}}$ in a local window. Then, the background score
is where $Z~Normal(0,1)$ and deviations from the norm are caused
by the strong ChIP-seq binding signals. \\


\includegraphics[scale=0.3]{Rplot08}\includegraphics[scale=0.3]{Rplot05_2}

Figure 3. (left) Kernel density plot of NormDiff for whole genome
shows a approximately normal distribution of the background reads.
(right) Q-Q Plot shows that NormDiff scores that represent strong
ChIP-seq binding signals are highly different from the background.


\section{Bibliography}

\bibliographystyle{plain}
\nocite{*}
\bibliography{References}

\end{document}
